src/main/java/core/basesyntax/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,207 @@
+                List<Driver> drivers = new ArrayList<>();
+                driversStatement.setLong(1, car.getId());
+                ResultSet driverResultSet = driversStatement.executeQuery();
+                while (driverResultSet.next()) {
+                    drivers.add(makeDriver(driverResultSet));
+                }
+                car.setDrivers(drivers);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to GET ALL cars from DB", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarString = "UPDATE cars SET car_id=?, manufacturer_id=?, car_model=?"
+                + " WHERE car_id=? AND deleted=false;";
+        String deleteLinksString = "DELETE FROM cars_drivers WHERE car_id=?;";
+        String newLinksString = "INSERT INTO cars_drivers (car_id, driver_id"
+                + ") VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement carUpdateStatement = connection.prepareStatement(updateCarString);
+                PreparedStatement deleteLinksStatement = connection
+                        .prepareStatement(deleteLinksString);
+                PreparedStatement newLinksStatement = connection.prepareStatement(newLinksString)) {
+            carUpdateStatement.setLong(1, car.getId());
+            carUpdateStatement.setLong(2, car.getManufacturer().getId());
+            carUpdateStatement.setString(3, car.getModel());
+            carUpdateStatement.setLong(4, car.getId());
+            carUpdateStatement.executeUpdate();
+            deleteLinksStatement.setLong(1,car.getId());
+            deleteLinksStatement.executeUpdate();
+            newLinksStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                newLinksStatement.setLong(2, driver.getId());
+                newLinksStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to UPDATE car in DB: "
+                    + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteString = "UPDATE cars SET deleted=true WHERE car_id=?;";
+        int result;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(deleteString)) {
+            statement.setLong(1, id);
+            result = statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to DELETE car from DB, id=" + id, e);
+        }
+        return result > 0;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarsByDriverString = "SELECT car_id FROM cars_drivers WHERE driver_id=?;";
```

Comment:
```suggestion
        String getCarsByDriverQuery = "SELECT car_id FROM cars_drivers WHERE driver_id=?;";
```

Reply:
What if the driver by this id is deleted?

Reply:
fixed